templates/connectors/using_gmail_connector/src/using_gmail_connector/main.bal

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/config;
+import ballerina/io;
+import wso2/gmail;
+
+// Setting the OAuth Client configuration.
+gmail:GmailConfiguration gmailConfig = {
+    oauthClientConfig: {
+        accessToken: config:getAsString("ACCESS_TOKEN"),
+        refreshConfig: {
+            refreshUrl: gmail:REFRESH_URL,
+            refreshToken: config:getAsString("REFRESH_TOKEN"),
+            clientId: config:getAsString("CLIENT_ID"),
+            clientSecret: config:getAsString("CLIENT_SECRET")
+        }
+    }
+};
+
+// Initializing a Gmail client.
+gmail:Client gmailClient = new (gmailConfig);
+
+string recipient = config:getAsString("RECIPIENT");
+string sender = config:getAsString("SENDER");
+string cc = config:getAsString("CC");
+string attachmentPath = config:getAsString("ATTACHMENT_PATH");
+string attachmentContentType = config:getAsString("ATTACHMENT_CONTENT_TYPE");
+string inlineImagePath = config:getAsString("INLINE_IMAGE_PATH");
+string inlineImageName = config:getAsString("INLINE_IMAGE_NAME");
+string imageContentType = config:getAsString("IMAGE_CONTENT_TYPE");
+string userId = config:getAsString("USER_ID");
+string emailId = "123456";
+
+public function main() {
+    //---Send Email---
```

Comment:
Rather than having these comments,  what if directly call the sendEmail readEmail Methods from main and do the relevant error handling within the functions itself? That way we don't need to add comments like Send Email within the function.

Reply:
Here the sent email id is used to do the read and delete. But can remove the error handling in readMail and deleteEmail and add them to the functions itself.

Reply:
Moved the error handling to functions.